src/find_library.cpp

```diff
@@ -70,4 +69,9 @@ std::string find_library_path(const std::string & library_name)
   return "";
 }
 
+std::string filename_for_library(const std::string & library_name)
```

Comment:
My personal preference for this was that it would look like 

```
std::string find_library_in_path(const std::string & library_name, const std::string & in_path)
{
  auto name = filename_for_library(library_name);
  auto final_path = in_path / name;
  if (final_path.is_file) {
    return final_path;
  }
  return "";
}
```

To be similar to `find_library_path` - but a more limited version for a _specific_ directory.

Reply:
That's fine with me as well. Can it throw an exception instead of returning an empty string?

Reply:
`find_library_path` doesn't throw an exception - I would think the two should act the same. I don't have a strong opinion which way that is, just that they should work the same

Reply:
@emersonknapp Please take another look if this is fine for you. I named it `path_for_library` instead since it doesn't do any finding.

Took me a while to figure out that cppcheck has issues with Gtests in namespaces.